### ISSUE TITLE
read/write with multiple backends

### ISSUE THREAD

**Issue by ryanSoley** [capitalone/rubicon-ml#357]
**Is your feature request related to a problem? Please describe**

sometimes it is desirable to write rubicon logs to multiple backends. currently, that'd be accomplished by instantiating a second `Rubicon` object and duplicating any logging code:

```python
rb_a = Rubicon(persistence="filesystem", root_dir="./local/root")
log_stuff(rb_a)  # calls to `log_` functions log to local filesystem

rb_b = Rubicon(persistence="filesystem", root_dir="s3://remote/bucket/root")
log_stuff(rb_b)  # calls to `log_` functions log to remote S3 filesystem
```

**Describe the solution you'd like**

we should create a new repository layer that can configure multiple backends at once for reading and writing. the client level entry point should look something like this:

```python
rb = Rubicon(composite_config=[
    {"persistence": "filesystem", "root_dir": "./local/root"},
    {"persistence": "filesystem", "root_dir": "s3://remote/bucket/root"},
])
log_stuff(rb)  # calls to `log_` functions log to both local filesystem and remote S3 filesystem
```

since the `Rubicon` object exposes arbitrary kwargs as `**storage_options` , we don't need to update the `Rubicon` object
to allow `composite_config` to be passed in. we just need to be aware that we need to retrieve it like this in the config: `self.storage_options.get("composite_config")`

at the repository layer, we'll need to create a new backend repo class (something like `CompositeRepository` in **rubicon_ml/repository/composite.py**). the current repo extensions are generally instantiated with a `root_dir` and `storage_options`, but this new one will be instantiated with other instantiated repos - perhaps something like this

```python
class CompositeRepository(BaseRepository):
    def __init__(self, repositories):
        self.repositories = repositories
        # other init ...
```

then, we'll need to update the [client **config.py**](https://github.com/capitalone/rubicon-ml/blob/main/rubicon_ml/client/config.py) to handle the `composite_config` argument properly. currently, for a single backend, we set `self.repository` to the single backend in the config's init. in the case of a composite backend, we'll need to create a repository for each of the entries in `composite_config` and append them to a list which can then be passed to `CompositeRepository`, which will ultimately be `self.repository` on the config

once the config is updated to properly instantiate a `CompositeRepository`, we'll need to override the functions in the [**Filesystem Helpers** section](https://github.com/capitalone/rubicon-ml/blob/main/rubicon_ml/repository/base.py#L42) to handle each of the repositories. in the case of functions that write data, we'll want to write to all repositories. possibly something like this:

```python
def _persist_bytes(self, bytes_data, path):
    for repo in self.repositories:
        repo._persist_bytes(bytes_data, path)
```

in the case of functions that read data, we'll want to read until we get a return value. we only need to read from one filesystem, not all of them since the data in them should be the same. possibly something like this:

```python
def _cat(self, path):
    for repo in self.repositories:
        try:
            result = repo._cat(path)
        except Exception as err:
            pass  # failed, move on to the next one
        else:
            return result  # passed, return the result of this one

    raise err  # all repos failed, raise the error
```

to summarize, I'd approach in this order
1. create a new file **composite.py** with `CompositeRepository` and its `__init__`
2. update **config.py** to properly handle the `composite_config` arg and instantiate a `CompositeRepository`
3. override the filesystem helpers in **composite.py**
